variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/repositories/VariantRepositoryImpl.java

```diff
@@ -104,8 +105,9 @@ public Long countByRegionsAndComplexFilters(List<Region> regions, List<VariantRe
 
     @Override
     public Set<String> findDistinctChromosomes() {
-        return new HashSet<>(mongoTemplate.getCollection(mongoTemplate.getCollectionName(VariantMongo.class))
-                .distinct(VariantMongo.CHROMOSOME_FIELD));
+        return StreamSupport.stream(mongoTemplate.getCollection(mongoTemplate.getCollectionName(VariantMongo.class))
+                                                 .distinct(VariantMongo.CHROMOSOME_FIELD, String.class).spliterator(),
+                                    false).collect(Collectors.toSet());
```

Comment:
any reasoning to do the extra step of streaming? otherwise it could be just the set construction from the result of the `distinct`, right?

Reply:
The return value from distinct is an iterable and can't be directly converted to a Set without streaming.